tests/ZonedDateTimeTest.php

```diff
@@ -482,4 +487,448 @@ public function providerCompareTo() : array
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $time = LocalTime::of(12, 34, 56, 987654321);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-01-23T12:34:56.987654321-08:00[America/Los_Angeles]', (string) $zonedDateTime->withTime($time));
+    }
+
+    public function testWithYear()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2012-02-20T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-02-20T12:34:56.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime->withYear(2000));
+    }
+
+    public function testWithMonth()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-02-25T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-01-25T12:34:56.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime->withMonth(1));
+    }
+
+    public function testWithDay()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-30T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-01-01T12:34:56.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime->withDay(1));
+    }
+
+    public function testWithHour()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-01-03T14:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-01-03T12:34:56.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime->withHour(12));
+    }
+
+    public function testWithMinute()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-03-28T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-03-28T12:30:56.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime->withMinute(30));
+    }
+
+    public function testWithSecond()
+    {
+        $timeZone = TimeZone::parse('America/Los_Angeles');
+        $localDateTime = '2000-03-25T12:34:56.123456789';
+        $localDateTime = LocalDateTime::parse($localDateTime);
+        $zonedDateTime = ZonedDateTime::of($localDateTime, $timeZone);
+
+        $this->assertSame('2000-03-25T12:34:56.123456789-08:00[America/Los_Angeles]', (string) $zonedDateTime->withSecond(56));
```

Comment:
Can you use another second value in `withSecond()`? It's the same as in the original date-time!